utils/hudi_clickhouse_sync.py

```
@@ -206,6 +241,55 @@ def close(self):
+        if not data_batch:
+            return
+
+        try:
+            # Attempt a delete of existing rows for the symbol+trade_date pairs in batch
+            symbols_and_dates = [(r[0], r[13]) for r in data_batch if r[0] and r[13]]
+            if symbols_and_dates:
+                delete_conditions = []
+                for symbol, trade_date in symbols_and_dates:
+                    delete_conditions.append(f"(symbol = '{symbol}' AND trade_date = '{trade_date}')")
+                if delete_conditions:
+                    delete_query = f"DELETE FROM {table_name} WHERE {' OR '.join(delete_conditions)}"
```

Ruff (0.14.10)

Possible SQL injection vector through string-based query construction

(S608)